src/FSharp.Control.TaskSeq.Test/TaskSeq.Let.Tests.fs

```diff
@@ -80,3 +80,73 @@ let ``CE taskSeq: use 'let!' with a non-generic task`` () =
     }
     |> verifyEmpty
     |> Task.map (fun _ -> value |> should equal 1)
+
+[<Fact>]
+let ``CE taskSeq: use 'let!' with Async`` () =
+    let mutable value = 0
+
+    taskSeq {
+        do value <- value + 1
+        let! _ = Async.Sleep 50
+        do value <- value + 1
+    }
+    |> verifyEmpty
+    |> Task.map (fun _ -> value |> should equal 2)
+
+[<Fact>]
+let ``CE taskSeq: use 'let!' with Async - mutables`` () =
+    let mutable value = 0
+
+    taskSeq {
+        do! async { value <- value + 1 }
+        do value |> should equal 1
+        let! x = async { return value + 1 }
+        do x |> should equal 2
+        do! Async.Sleep 50
+        do! async { value <- value + 1 }
+        do value |> should equal 2
+        let! ret = async { return value + 1 }
+        do value |> should equal 2
+        do ret |> should equal 3
+        yield x + ret // eq 5
+    }
+    |> TaskSeq.exactlyOne
+    |> Task.map (should equal 5)
+
+[<Fact>]
+let ``CE taskSeq: use 'let!' with all kinds of overloads at once`` () =
+    let mutable value = 0
+
+    // this test should be expanded in case any new overload is added
+    // that is supported by `let!`, to ensure the high/low priority
+    // overloads still work properly
+    taskSeq {
+        let! a = task { // eq 1
+            do! Task.Delay 10
+            do value <- value + 1
+            return value
+        }
+
+        let! b =  // eq 2
+            task {
+                do! Task.Delay 50
+                do value <- value + 1
+                return value
+            }
+            |> ValueTask<int>
+
+        let! c = ValueTask<_>(4) // valuetask that completes immediately
+        let! _ = Task.Factory.StartNew(fun () -> value <- value + 1) // non-generic Task with side effect
```

Comment:
I held of on mentioning this earlier but it comes up in another issue....

This was the original .NET 4.0 set of Thread Pool helpers. All usage of `Task.Factory` has been shifted to `Task` helpers. In this specific case, the canonical way to express this is as a `Task.Run`. The `Run` verb is associated with a) starting stuff b) in the thread pool

Reply:
The overloads and signature are very painful from F# though, hence a helper like this is normally required to make it palatable https://github.com/jet/propulsion/blob/4d62f9d8afc3ab87273a0765e09a065ea1e66ad9/src/Propulsion/Internal.fs#L90